server/routes.go

```diff
@@ -1539,6 +1572,18 @@ func (s *Server) ChatHandler(c *gin.Context) {
 		return
 	}
 
+	if req.DryRun {
```

Comment:
First pass at this and have some thoughts:
- I'm wondering if we could wrap this under the existing "options" parameter although that is meant for the model options and I'm not a fan of having the two conflated - it's something I'll try it out
- With this method we have to load the full model into the vRAM _and_ use the scheduler due to needing the tokenizer as well as the truncated content of the messages.
- There is a world where we can side load the model without the vRAM (like in the [tokenize draft PR](https://github.com/ollama/ollama/pull/8106/files#diff-829f94fd58de777429c6c25e68de745804cb600031e92995007fa08d7af32078R588-R601)). 
- We'd still have to refactor doing the truncation based on context length not in the runner and I think that's where we start bleeding scope.

My take is:
- Cleanup this PR - keep scope small
- Figure out more long term model loading + swapping for quick interactions vs loading into vRAM and have a shared interface for those common patterns

Reply:
(also to address streaming)